presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -296,16 +305,65 @@ public ConnectorSplitSource getSplits(
                 splitSchedulingContext.schedulerUsesHostAddresses(),
                 layout.isPartialAggregationsPushedDown());
 
-        HiveSplitSource splitSource = computeSplitSource(splitSchedulingContext, table, session, hiveSplitLoader);
+        HiveSplitSource splitSource = computeSplitSource(splitSchedulingContext, table, session, hiveSplitLoader, ratio);
         hiveSplitLoader.start(splitSource);
 
         return splitSource;
     }
 
+    private double getSplitScanRatio(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            HiveTableLayoutHandle layout,
+            TransactionalMetadata metadata,
+            List<HivePartition> partitions)
+    {
+        double ratio = 1.0;
```

Comment:
we don't really use this variable in the logic. We either return 1.0 for unsupported cases or we return some computation that doesn't use any previously set value.  I would get rid of this variable and inline all the returns instead.  It makes it easier to see t's not really changed.